cryptocurrency/cryptocurrency

```diff
@@ -0,0 +1,306 @@
+  base=$(echo $base | tr /a-z/ /A-Z/)
+  if [[ $(checkValidCurrency $base) == "1"  ]]; then
+    unset base
+    echo "Invalid base currency"
+    getBase
+  fi
+}
+
+## Checks base currency from the user and validates it with all the possible currency
+## types available on the API (requires argument)
+checkBase()
+{
+  base=$1
+  base=$(echo $base | tr /a-z/ /A-Z/)
+  if [[ $(checkValidCurrency $base) == "1" ]]; then
+    unset base
+    echo "Invalid base currency"
+    exit 1
+  fi
+}
+
+# Matches the symbol to the appropriate ids regarding the API calling.
+transformBase()
+{
+  case "$base" in 
+    "ETH") reqId="ethereum" ;;
+    "BTC") reqId="bitcoin" ;;
+    "XRP") reqId="ripple" ;;
+    "LTC") reqId="litecoin" ;;
+    "XEM") reqId="nem" ;;
+    "ETC") reqId="ethereum-classic" ;;
+    "DASH") reqId="dash" ;;
+    "MIOTA") reqId="iota" ;;
+    "XMR") reqId="monero" ;;
+    "STRAT") reqId="stratis" ;;
+  esac
+}
+
+## Grabs the exchange to currency from the user and validates it with all the possible currency
+## types available on the API and guides user through input (doesnt take in arguments)
+getExchangeTo()
+{
+  echo -n "What currency to exchange to: "
+  read -r exchangeTo
+  exchangeTo=$(echo $exchangeTo | tr /a-z/ /A-Z/)
+  if [[ $(checkTargetCurrency $exchangeTo) == "1" ]]; then
+    echo "Invalid exchange currency"
+    unset exchangeTo
+    getExchangeTo
+  fi
+}
+
+## Grabs the exchange to currency from the user and validates it with all the possible currency
+## types available on the API (requires arguments)
+checkExchangeTo()
+{
+  exchangeTo=$1
+  exchangeTo=$(echo $exchangeTo | tr /a-z/ /A-Z/)
+  if [[ $(checkTargetCurrency $exchangeTo) == "1" ]]; then
+    echo "Invalid exchange currency"
+    unset exchangeTo
+    exit 1
+  fi
+}
+
+## Get the amount that will be exchanged and validate that the user has entered a number (decimals are allowed)
+## doesnt take in argument, it guides user through input
+getAmount()
+{
+  echo -n "What is the amount being exchanged: "
+  read -r amount
+  if [[ ! "$amount" =~ ^[0-9]+(\.[0-9]+)?$ ]]; then
+    echo "The amount has to be a number"
+    unset amount
+    getAmount
+  fi
+}
+
+## Get the amount that will be exchanged
+## validate that the user has entered a number (decimals are allowed and requires argument)
+checkAmount()
+{
+  amount=$1
+  if [[ ! "$amount" =~ ^[0-9]+(\.[0-9]+)?$ ]]; then
+    echo "The amount has to be a number"
+    unset amount
+    exit 1
+  fi
+}
+
+checkInternet()
+{
+  httpGet google.com > /dev/null 2>&1 || { echo "Error: no active internet connection" >&2; return 1; } # query google with a get request
+}
+
+## Grabs the exchange rate and does the math for converting the currency
+convertCurrency()
+{
+  exchangeTo=$(echo "$exchangeTo" | tr '[:upper:]' '[:lower:]')
+  exchangeRate=$(httpGet "https://api.coinmarketcap.com/v1/ticker/$reqId/?convert=$exchangeTo" | pcregrep -o2 "(\"price_$exchangeTo\")\s*:\s*.([0-9]*[.][0-9]*)") > /dev/null
```

Comment:
Because afaik curl buffers the output and outputs the chunks by default. For this particular case this doesn't make any sense right? See [here](https://curl.haxx.se/docs/manpage.html#-N).